### PR TITLE
WFLY-13904 Support ProtoStream marshalling of distributed web session attributes

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-13904

Analysis document: https://github.com/wildfly/wildfly-proposals/pull/453

### CODE REVIEW

**Comment by pferraro** [comment]
@rachmatowicz @rhusar Please review.

File: clustering/weld/core/src/test/java/org/wildfly/clustering/weld/annotated/slim/unbacked/UnbackedMemberIdentifierMarshallerTestCase.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.wildfly.clustering.weld.annotated.slim.unbacked;
+
+/**
+ * @author Paul Ferraro
+ */
+public class UnbackedMemberIdentifierMarshallerTestCase {
+
```
[the remark this answers is not in the document]
Comment (pferraro) [reply clustering/weld/core/src/test/java/org/wildfly/clustering/weld/annotated/slim/unbacked/UnbackedMemberIdentifierMarshallerTestCase.java:28]: This was not meant to be committed. Thanks for catching this. :)

**Review by rhusar — APPROVED** [review]
I don't see any issues, looks good to me! Just adding a couple optional comments.

File: clustering/web/extension/src/main/java/org/wildfly/extension/clustering/web/DistributableWebSchema.java [hunk 2]
```diff
@@ -32,9 +32,10 @@
  */
 public enum DistributableWebSchema implements Schema<DistributableWebSchema> {
     VERSION_1_0(1, 0), // WildFly 17
-    VERSION_2_0(2, 0), // WildFly 18
+    VERSION_2_0(2, 0), // WildFly 18-21
+    VERSION_3_0(3, 0), // WildFly 22
```
[the remark this answers is not in the document]
Comment (pferraro) [reply clustering/web/extension/src/main/java/org/wildfly/extension/clustering/web/DistributableWebSchema.java:36]: Now you know for how long I've been working on this branch... :)

File: clustering/marshalling/spi/src/main/java/org/wildfly/clustering/marshalling/spi/ByteBufferMarshalledValue.java [hunk 3]
```diff
@@ -65,7 +68,9 @@ public synchronized ByteBuffer getBuffer() throws IOException {
         ByteBuffer buffer = this.buffer;
         if (buffer != null) return buffer;
         if (this.object == null) return null;
-        return this.marshaller.write(this.object);
+        ByteBuffer result = this.marshaller.write(this.object);
+        LOGGER.debugf("Marshalled size of %s(%s) = %d bytes", this.object.getClass().getCanonicalName(), this.object, result.limit() - result.arrayOffset());
```
[the remark this answers is not in the document]
Comment (pferraro) [reply clustering/marshalling/spi/src/main/java/org/wildfly/clustering/marshalling/spi/ByteBufferMarshalledValue.java:72]: DEBUG is intentional.  This is not here for purposes of tracing.

File: clustering/web/extension/src/main/java/org/wildfly/extension/clustering/web/DistributableWebSchema.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (rhusar) [reply clustering/web/extension/src/main/java/org/wildfly/extension/clustering/web/DistributableWebSchema.java]: @pferraro Oh yes... that was clear :-)

**Comment by pferraro** [comment]
I'm going to kill the remaining CI jobs - I only needed to verify that the last 2 commits (which modify comments only) pass checkstyle...

**Pull request merged**